src/python/pants/backend/openapi/codegen/python/extra_fields.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2024 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+import itertools
+from pants.backend.openapi.target_types import OpenApiDocumentGeneratorTarget, OpenApiDocumentTarget
+from pants.backend.python.target_types import PrefixedPythonResolveField
+from pants.engine.target import BoolField, DictStringToStringField, StringField
+
+
+class OpenApiPythonGeneratorNameField(StringField):
+    alias = "python_generator_name"
+    required = False
+    help = "Python generator name"
```

Comment:
> nothing is stopping an end user for typing python_generator_name="rust"

I don't think this is the problem we need to solve. Similarly, one could use `python_sources(sources=["*.rs"])`, we don't want to validate such a thing. Moreover `openapi-generator list` doesn't tell us which language it generates, which makes sense because the generator can actually generate multiple different kinds of files, like .py, .txt, .yaml. Anyway, the validation will create more problems than it will solve.

>  simply misspelling the value

Yes, this is a good idea, we can use `openapi-generator list` for simple set validation.

Reply:
Added validation with `OpenAPIGeneratorNames`

Reply:
> Moreover openapi-generator list doesn't tell us which language it generates

I rather take it more like the generator _targets_ a language + tooling, it does generate support files that are not in the target language, this is because it aims at generating a self-contained buildable product. And this isn't 100% correct either as it also can generate docs and Apache httpd configuration files so I understand that hardcoding an assumption into it their naming convention could be problematic. Anyway, at least validating that the chosen name is an accepted one before running the tool gives a better UX.